contributors/devel/sig-instrumentation/logging.md

```diff
@@ -1,37 +1,78 @@
-## Logging Conventions
+## Logging
 
-The following conventions for the klog levels to use.
-[klog](http://godoc.org/github.com/kubernetes/klog) is globally preferred to
-[log](http://golang.org/pkg/log/) for better runtime control.
+This document provides an overview of  the recommended way to develop and implement
+logging for components of Kubernetes.
+For [k/kubernetes](https://github.com/kubernetes/kubernetes) these conventions should be seen
+as a strong recommendation, however any Kubernetes or external project is welcome to follow
+the guidelines below. Most of the ideas can be applied more broadly.
 
-Shared libraries, such as `client-go`, should not use `klog.ErrorS()` and `klog.InfoS()`,
+### Logging in Kubernetes
+
+Kubernetes project uses [klog](https://github.com/kubernetes/klog) for logging.
+This library was created as a permanent fork of glog, as the original project was no longer developed. 
+However, should not be used by external projects as it is mostly developed to maintain backward compatibility.
+With the introduction of [Structured Logging](https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging), Kubernetes is being migrated to use [logr](https://github.com/go-logr/logr) as its logging interface. Any changes in klog are done to ensure a smooth transition to `logr`.
+``
+
+For projects that want to integrate with Kubernetes or write same log format we recommend using [klogr](https://github.com/kubernetes/klog/tree/main/klogr), reimplementation of klog format that exposes [logr](https://github.com/go-logr/logr) interface.
+
+### Logging Conventions
+
+### When not to log?
+
+Shared libraries, such as [`client-go`](https://github.com/kubernetes/client-go), should _not_
+write logs themselves,
 but just return `error`, because client libraries may be used in CLI UIs that wish to control output.
```

Comment:
```suggestion
### When not to log?

Shared libraries, such as [`client-go`](https://github.com/kubernetes/client-go), should _not_
log errors themselves
but just return `error`, because client libraries may be used in CLI UIs that wish to control output.
```